components/EditTask.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react';
+import { View, StyleSheet, TextInput, Text, Button } from 'react-native';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { useDate } from '../providers/DateProvider';
+import useActivity from '../hooks/useActivity';
+import { useCitizen } from '../providers/CitizenProvider';
+import { router } from 'expo-router';
+
+type EditTaskButtonProps = {
+  title: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  activityId: number;
+  isCompleted: boolean;
+};
+
+type SubmitProps = {
+  title: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  date: Date;
+};
+
+const EditTask = ({
+  title,
+  description,
+  startTime,
+  endTime,
+  activityId,
+  isCompleted,
+}: EditTaskButtonProps) => {
+  const [form, setForm] = useState<SubmitProps>({
+    title: title,
+    description: description,
+    startTime: startTime,
+    endTime: endTime,
+    date: new Date(endTime),
+  });
+
+  const { selectedDate } = useDate();
+  const { citizenId } = useCitizen();
+  const { updateActivity } = useActivity({ date: selectedDate });
+
+  const handleInputChange = (
+    field: keyof SubmitProps,
+    value: string | Date
+  ) => {
+    setForm((prevData) => ({
+      ...prevData,
+      [field]: value,
+    }));
+  };
+
+  const handleSubmit = async () => {
+    const startTimeHHMM = form.startTime.toLocaleTimeString('it-IT', {
+      hour: '2-digit',
+      minute: '2-digit',
+    });
+    const endTimeHHMM = form.endTime.toLocaleTimeString('it-IT', {
+      hour: '2-digit',
+      minute: '2-digit',
+    });
+
+    await updateActivity.mutateAsync({
+      activityId: activityId,
+      citizenId: citizenId,
+      date: form.date.toDateString(),
+      name: form.title,
+      description: form.description,
+      startTime: startTimeHHMM,
+      endTime: endTimeHHMM,
+      isCompleted: isCompleted,
+    });
+    router.back();
+  };
+
+  return (
+    <View style={styles.modalContent}>
+      <Text style={styles.modalTitle}>Ændre Aktivitet</Text>
+      <TextInput
+        value={form.title}
+        style={styles.input}
+        onChangeText={(text) => setForm((prev) => ({ ...prev, title: text }))}
+      />
+      <TextInput
+        value={form.description}
+        style={styles.input}
+        onChangeText={(text) =>
+          setForm((prev) => ({ ...prev, description: text }))
+        }
+      />
+      <View style={styles.pickerContainer}>
+        <Text style={styles.header}>Vælg start tid</Text>
+        <DateTimePicker
+          mode="time"
+          value={form.startTime}
+          is24Hour={true}
+          display="default"
+          onChange={(_event, selectedDate) => {
+            if (!selectedDate) return;
+            handleInputChange('startTime', selectedDate);
+          }}
+          style={styles.timePicker}
+        />
+      </View>
+
+      <View style={styles.pickerContainer}>
```

Comment:
Shouldn't we want to ensure endTime is later than startTime somewhere?

Reply:
We should, it's a general theme ive seen but yes we should change it